test/resource_test.rb

```diff
@@ -415,6 +415,26 @@ def test_permissions
     end
   end
 
+  def test_namespace
+    previous_namespace = Semian.namespace
+
+    resource = create_resource :testing, permissions: 0o600, tickets: 1
+    key = resource.key
+    Semian.destroy(:testing)
+
+    resource = create_resource :testing, permissions: 0o600, tickets: 1
+    assert_equal key, resource.key
```

Comment:
Had to look at why you're not asserting a match / equality with a human-readable string.. Looks like `resource.key` is hex string of a SHA digest 👍